neon_utterance_translator_plugin/__init__.py

```diff
@@ -35,30 +35,43 @@ class UtteranceTranslator(UtteranceTransformer):
 
     def __init__(self, name="utterance_translator", priority=5):
         super().__init__(name, priority)
-        self.language_config = self.config.get("language") or {}
         self.lang_detector = OVOSLangDetectionFactory.create()
         self.translator = OVOSLangTranslationFactory.create()
 
     def transform(self, utterances, context=None):
         metadata = []
-        internal_lang = self.language_config["internal"]
+        lang = context.get('lang') or 'en-us'
+        internal_lang = context.get('internal') or 'en-us'
+        supported_langs = context.get('supported_langs') or ['en']
+        was_translated = False
         for idx, ut in enumerate(utterances):
             try:
                 original = ut
                 detected_lang = self.lang_detector.detect(original)
-                LOG.debug(f"Detected language: {detected_lang}")
-                if detected_lang != internal_lang.split("-")[0]:
-                    utterances[idx] = self.translator.translate(original, internal_lang)
+                if detected_lang != lang.split('-', 1)[0]:
+                    LOG.warning(f"Specified lang: {lang} but detected {detected_lang}")
+                else:
+                    LOG.debug(f"Detected language: {detected_lang}")
+                if detected_lang not in supported_langs:
+                    LOG.warning(f"There is no: {lang} in supported languages. "
+                                f"Utterance will be translated to English")
+                    utterances[idx] = self.translator.translate(
+                        original,
+                        internal_lang,
+                        detected_lang)
```

Comment:
I think we'll want to use `lang` if it was specified in context here; detection can be inaccurate on short utterances in particular.